tests/unit/test_compress.py

```diff
@@ -0,0 +1,276 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import gzip
+import io
+
+import pytest
+
+from botocore.compress import maybe_compress_request
+from botocore.config import Config
+from tests import mock
+
+
+def _op_with_compression():
+    op = mock.Mock()
+    op.request_compression = {'encodings': ['gzip']}
+    op.has_streaming_input = False
+    return op
+
+
+def _op_with_multiple_compressions():
+    op = mock.Mock()
+    # TODO: Update second value to a different encoding. Currently only gzip
+    # is supported.
+    op.request_compression = {'encodings': ['gzip', 'gzip']}
+    op.has_streaming_input = False
+    return op
+
+
+def _op_unknown_compression():
+    op = mock.Mock()
+    op.request_compression = {'encodings': ['foo']}
+    op.has_streaming_input = None
```

Comment:
Why is this one `None` while the others are `False`?

Reply:
I think this must've been a mistake. It'll be cleaned up in the next commit.